### PR TITLE
Improve readability of DemuxPostProcessingAPI

### PR DESCRIPTION
**Pull request by seallard**
## Description

This PR improves the readability of the `DemuxPostProcessingAPI`, specifically the `finish_flow_cell_temp` function and the ones  it calls.

### Fixed
- Improve readability of `DemuxPostProcessingAPI`

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by seallard** [comment]
Deployed and tested in stage

File: cg/cli/demultiplex/finish.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def finish_flow_cell(
 @click.pass_obj
 def store_sequencing_metrics(context: CGConfig, flow_cell_name: str):
     demux_post_processing_api: DemuxPostProcessingAPI = DemuxPostProcessingAPI(config=context)
-    demux_post_processing_api.finish_flow_cell_temp(flow_cell_name=flow_cell_name)
+    demux_post_processing_api.finish_flow_cell_temp(flow_cell_directory_name=flow_cell_name)
```
Comment (henrikstranneheim) [inline cg/cli/demultiplex/finish.py:64]:
```suggestion
    demux_post_processing_api.finish_flow_cell_temp(flow_cell_directory_name=flow_cell_directory_name)
```
Would make it even more consistent

File: cg/meta/demultiplex/demux_post_processing.py [hunk 2]
```diff
@@ -288,41 +280,44 @@ def file_exists_in_latest_version_for_bundle(
...
+            raise FlowCellError(f"Flow cell directory not found: {flow_cell_directory}")
+
         try:
             flow_cell: FlowCellDirectoryData = FlowCellDirectoryData(
                 flow_cell_path=flow_cell_directory, bcl_converter=bcl_converter
             )
             return flow_cell
-
-        except FlowCellError:
+        except FlowCellError as e:
```
Comment (henrikstranneheim) [inline cg/meta/demultiplex/demux_post_processing.py:306]:
```suggestion
        except FlowCellError as error:
```
or exception? e is so ambigous

**Review by henrikstranneheim — APPROVED** [review]
Nice 👍

File: cg/meta/demultiplex/demux_post_processing.py [hunk 3]
```diff
@@ -71,78 +70,77 @@ def transfer_flow_cell(
...
-        5. Create sequencing metrics.
-        6. Update samples in status db with read counts and sequencing date.
-        """
+        Raises:
+            FlowCellError: If the flow cell directory or the data it contains is not valid.
 
-        LOG.info(f"Finish flow cell {flow_cell_name}")
+        Note:
+            This method does not return anything but updates the status and housekeeper databases
+            with data from the flow cell directory.
```
Comment (diitaz93) [inline cg/meta/demultiplex/demux_post_processing.py:91]:
```suggestion
        Note:
            If successful, returns None and updates the status and housekeeper databases
            with data from the flow cell directory.
```

Comment (diitaz93) [reply cg/meta/demultiplex/demux_post_processing.py:91]: Even maybe change the `Note:` for `Return:`?

**Comment by seallard** [comment]
Deployed and tested in stage.

File: cg/meta/demultiplex/demux_post_processing.py [hunk 4]
```diff
@@ -151,39 +149,34 @@ def is_flow_cell_directory_valid(self, flow_cell_directory: Path) -> bool:
...
-
             LOG.info(f"Updating sample {sample_id} with read count {sample_read_count}")
-
             sample.reads = sample_read_count
 
-        self.status_db.session.commit()
-
     def add_flow_cell_data_to_housekeeper(
         self, flow_cell_name: str, flow_cell_directory: Path
     ) -> None:
```
Comment (diitaz93) [inline cg/meta/demultiplex/demux_post_processing.py:179]: Why did you delete the doctring?

File: cg/meta/demultiplex/demux_post_processing.py [hunk 5]
```diff
@@ -151,39 +149,34 @@ def is_flow_cell_directory_valid(self, flow_cell_directory: Path) -> bool:
...
 
+    def update_sample_read_counts(self, sample_internal_ids: List[str]) -> None:
+        """Update samples in status db with the sum of all read counts for the sample in the sequencing metrics table."""
         for sample_id in sample_internal_ids:
-            LOG.info(f"Updating read count for sample {sample_id}")
-            sample: Optional[Sample] = self.status_db.get_sample_by_internal_id(
-                internal_id=sample_id
-            )
+            self.update_single_sample_read_count(sample_id)
+        self.status_db.session.commit()
```
Comment (diitaz93) [inline cg/meta/demultiplex/demux_post_processing.py:165]: Nice 👍

File: cg/meta/demultiplex/demux_post_processing.py [hunk 6]
```diff
@@ -71,78 +70,77 @@ def transfer_flow_cell(
...
+            parsed_flow_cell=parsed_flow_cell
         )
 
-        self.update_sample_read_counts(sample_internal_ids=sample_ids)
+        self.update_sample_read_counts(sample_internal_ids=flow_cell_sample_ids)
+
+    def add_sample_lane_sequencing_metrics_for_flow_cell(
+        self, flow_cell_directory: Path, bcl_converter: str
+    ) -> None:
+        sample_lane_sequencing_metrics: List[
```
Comment (diitaz93) [inline cg/meta/demultiplex/demux_post_processing.py:126]: add docstring

**Review by diitaz93 — COMMENTED** [review]
👍

File: cg/meta/demultiplex/demux_post_processing.py (hunk 4 above)
Comment (ChrOertlin) [reply cg/meta/demultiplex/demux_post_processing.py:179]: I think it is due to that in this case the docstring is a repetition of the function name and violates DRY in this way as it adds no other knowledge than what can be understood from the function name.

I mention this in the coding guidelines in Atlas. That being said, we have not yet discussed on how to implement these things. So I am not sure what would be right to do at this point, i.e. keeping or deleting such docstrings

File: cg/meta/demultiplex/demux_post_processing.py [hunk 7]
```diff
@@ -71,78 +70,77 @@ def transfer_flow_cell(
...
+            - Stores the flow cell in the status database
+            - Stores sequencing metrics in the status database
+            - Updates sample read counts in the status database
+            - Stores the flow cell data in the housekeeper database
 
-        self.status_db.session.add_all(sample_lane_sequencing_metrics)
-        self.status_db.session.commit()
-        LOG.info(f"Added sample lane sequencing metrics to status database for: {flow_cell_name}")
+        Args:
+            flow_cell_directory_name (str): The name of the flow cell directory to be finalized.
```
Comment (ChrOertlin) [inline cg/meta/demultiplex/demux_post_processing.py:84]: finalised (British spelling)

File: cg/meta/demultiplex/demux_post_processing.py [hunk 8]
```diff
@@ -71,78 +70,77 @@ def transfer_flow_cell(
 
         LOG.info(f"Flow cell added: {flow_cell}")
 
-    def add_sample_lane_sequencing_metrics_for_flow_cell(self, flow_cell_name: str):
-        """Add sample lane sequencing metrics to status database."""
-        flow_cell_dir: Path = Path(self.demux_api.out_dir, flow_cell_name)
-        bcl_converter: str = self.get_bcl_converter(flow_cell_name=flow_cell_name)
+    def finish_flow_cell_temp(self, flow_cell_directory_name: str) -> None:
+        """Finalize the flow cell once the temporary demultiplexing process is completed.
```
Comment (ChrOertlin) [inline cg/meta/demultiplex/demux_post_processing.py:74]: Finalise
